src/main/java/org.wso2.apim.monetization/impl/StripeMonetizationDAO.java

```diff
@@ -582,7 +591,17 @@ public int addBESharedCustomer(MonetizationSharedCustomer sharedCustomer) throws
             ps.setInt(3, sharedCustomer.getTenantId());
             ps.setString(4, sharedCustomer.getSharedCustomerId());
             ps.setInt(5, sharedCustomer.getParentCustomerId());
-            id = ps.executeUpdate();
+            ps.executeUpdate();
+            ResultSet set = ps.getGeneratedKeys();
+            if (set.next()) {
+                id = set.getInt(1);
+            } else {
+                String errorMessage = "Failed to set ID of the shared customer : " + sharedCustomer.getId() +
+                        " , tenant ID : " + sharedCustomer.getTenantId() + " , application ID : " +
+                        sharedCustomer.getApplicationId();
+                log.error(errorMessage);
+                throw new StripeMonetizationException(errorMessage);
```

Comment:
Do we need to log the error message here as we throw the Exception with the error message?
This may cause duplicate error logging.

Reply:
Fixed : https://github.com/wso2-extensions/wso2-am-stripe-plugin/pull/5/commits/5f0a65ec9820d2a1e9a054bed35199fcb49c10b3